File: python/vnx-py-011/sql_injection.py
# VNX-PY-011: Python SQL injection
import sqlite3

conn = sqlite3.connect(":memory:")
cursor = conn.cursor()

user_input = "admin"
cursor.execute(f"SELECT * FROM users WHERE name = '{user_input}'")
cursor.execute("SELECT * FROM users WHERE id = %s" % user_input)

File: java/vnx-java-003/SqlInjection.java
import java.sql.*;

// VNX-JAVA-003: SQL injection via string concatenation
public class SqlInjection {
    public User findUser(Connection conn, String username) throws SQLException {
        Statement stmt = conn.createStatement();
        ResultSet rs = stmt.executeQuery("SELECT * FROM users WHERE name = '" + username + "'");
        return null;
    }
}
